pkg/dispatchcli/cmd/create_policy.go

```diff
@@ -0,0 +1,97 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/vmware/dispatch/pkg/dispatchcli/i18n"
+
+	policy "github.com/vmware/dispatch/pkg/identity-manager/gen/client/policy"
+	models "github.com/vmware/dispatch/pkg/identity-manager/gen/models"
+)
+
+var (
+	createPolicyLong = i18n.T(`Create a dispatch identity manager policy`)
+
+	createPolicyExample = i18n.T(`  dispatch create policy test_policy -s xyz@example.com -a get,update -r image,api,function`)
+
+	subject  string
+	action   string
+	resource string
+)
+
+// NewCmdCreatePolicy creates command responsible for dispatch policy creation
+func NewCmdCreatePolicy(out io.Writer, errOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     i18n.T(`policy POLICY_NAME --subject SUBJECTS --action ACTIONS --resource RESOURCES`),
+		Short:   i18n.T("Create policy"),
+		Long:    createPolicyLong,
+		Example: createPolicyExample,
+		Args:    cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := createPolicy(out, errOut, cmd, args)
+			CheckErr(err)
+		},
+	}
+
+	cmd.Flags().StringVarP(&subject, "subject", "s", "*", "subjects of policy rule, separated by comma")
+	cmd.Flags().StringVarP(&action, "action", "a", "*", "actions of policy rule, separated by comma")
+	cmd.Flags().StringVarP(&resource, "resource", "r", "*", "resources of policy rule, separated by comma")
+	return cmd
+}
+
+// CallCreatePolicy makes the api call to create a policy
+func CallCreatePolicy(p interface{}) error {
+	client := identityManagerClient()
+	policyModel := p.(*models.Policy)
+
+	params := &policy.AddPolicyParams{
+		Body:    policyModel,
+		Context: context.Background(),
+	}
+
+	created, err := client.Policy.AddPolicy(params, GetAuthInfoWriter())
+	if err != nil {
+		return formatAPIError(err, params)
+	}
+	*policyModel = *created.Payload
+	return nil
+}
+
+func createPolicy(out, errOut io.Writer, cmd *cobra.Command, args []string) error {
+
+	policyName := args[0]
+	policyRules := models.PolicyRules{
+		&models.Rule{
+			Subjects:  strings.Split(subject, ","),
```

Comment:
If you use `StringSliceP` then you don't have to split this.

Reply:
fixed